internal/cmd/skupper/link/nonkube/link_status.go

```diff
@@ -2,28 +2,129 @@ package nonkube
 
 import (
 	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/skupperproject/skupper/api/types"
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
+	"github.com/skupperproject/skupper/internal/nonkube/client/fs"
+	"github.com/skupperproject/skupper/pkg/utils/validator"
 	"github.com/spf13/cobra"
 )
 
 type CmdLinkStatus struct {
-	CobraCmd  *cobra.Command
-	Flags     *common.CommandLinkStatusFlags
-	Namespace string
-	siteName  string
+	linkHandler *fs.LinkHandler
+	siteHandler *fs.SiteHandler
+	CobraCmd    *cobra.Command
+	Flags       *common.CommandLinkStatusFlags
+	namespace   string
+	linkName    string
 }
 
 func NewCmdLinkStatus() *CmdLinkStatus {
 	return &CmdLinkStatus{}
 }
 
 func (cmd *CmdLinkStatus) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.linkHandler = fs.NewLinkHandler(cmd.namespace)
+	cmd.siteHandler = fs.NewSiteHandler(cmd.namespace)
+}
+
+func (cmd *CmdLinkStatus) ValidateInput(args []string) []error {
+	var validationErrors []error
+	opts := fs.GetOptions{LogWarning: false}
+	resourceStringValidator := validator.NewResourceStringValidator()
+
+	// Validate arguments name if specified
+	if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if len(args) == 1 {
+		if args[0] == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("link name must not be empty"))
+		} else {
+			ok, err := resourceStringValidator.Evaluate(args[0])
+			if !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("link name is not valid: %s", err))
+			} else {
+				cmd.linkName = args[0]
+			}
+		}
+	}
+	// Validate that there is a link with this name in the namespace
+	if cmd.linkName != "" {
+		link, err := cmd.linkHandler.Get(cmd.linkName, opts)
+		if link == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("link %s does not exist in namespace %s", cmd.linkName, cmd.namespace))
+		}
+	}
+	return validationErrors
 }
 
-func (cmd *CmdLinkStatus) ValidateInput(args []string) []error { return nil }
-func (cmd *CmdLinkStatus) InputToOptions()                     {}
 func (cmd *CmdLinkStatus) Run() error {
-	return fmt.Errorf("command not supported by the selected platform")
+	opts := fs.GetOptions{RuntimeFirst: true, LogWarning: true}
+	if cmd.linkName == "" {
+		links, err := cmd.linkHandler.List()
+		if links == nil || err != nil {
+			fmt.Println("no links found:")
```

Comment:
The output when no link is defined has:

```
no links found:
failed to read directory: open /home/fgiorget/.local/share/skupper/namespaces/west/runtime/resources/links: no such file or directory
```

If would be nice to add an extra condition to ignore such errors.

Reply:
@fgiorgetti what would you like here?  

don't show any output if links not found?     This be the default.   have a flag --detail  or --errors that will show above logs.   Do you want that for all status commands?

Reply:
I believe that in case there is no links defined under `runtime/resources` it should simply say: `no links found`.

Reply:
For sites where no links are defined, along with the error, it also says: "Site not initialized yet".

Reply:
Now if no links are found will only show:
"No links found"